### PR TITLE
Add ability to use challenges other than Route53 DNS

### PR DESCRIPTION
**Pull request by felddy**
#  #

## 🗣 Description ##

- Add the ability to use other Certbot challenges other than the default Route53.
- Upstream changes to implement `ghcr.io` container repository.

By adding the `--no-dns-route53` as the first parameter to the container, it will disable the default Route53 challenge.
This allows other challenges to be used.  e.g.; `--manual` 

## 💭 Motivation and context ##

We have a use case where we need to get certificates for domains that are not (yet) hosted in Route53.  

## 🧪 Testing ##

@dav3r is testing this with an external domain.
- [x] @dav3r says "it's gud".

Tested in CI.

## ✅ Checklist ##

* [x] This PR has an informative and human-readable title.
* [x] Changes are limited to a single goal - _eschew scope creep!_
* [x] All relevant type-of-change labels have been added.
* [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
* [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
* [x] All relevant repo and/or project documentation has been updated
      to reflect the changes in this PR.
* [x] All new and existing tests pass.

### CODE REVIEW

**Comment by dav3r** [comment]
Successful test!

```
❱ docker-compose -f my-docker-compose.yml run certboto --no-dns-route53 certonly --manual --preferred-challenges dns -d mysuperfakedomain.gov -d mail.mysuperfakedomain.gov -d www.mysuperfakedomain.gov
Syncing certbot configs from my-certificates-bucket
Rebuilding symlinks in /etc/letsencrypt
Route53 DNS challenge disabled by --no-dns-route53 flag
Running: certbot certonly --manual --preferred-challenges dns -d mysuperfakedomain.gov -d mail.mysuperfakedomain.gov -d www.mysuperfakedomain.gov
Saving debug log to /var/log/letsencrypt/letsencrypt.log
Plugins selected: Authenticator manual, Installer None
Requesting a certificate for mysuperfakedomain.gov and 2 more domains
Performing the following challenges:
dns-01 challenge for mail.mysuperfakedomain.gov
dns-01 challenge for mysuperfakedomain.gov
dns-01 challenge for www.mysuperfakedomain.gov

- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
Please deploy a DNS TXT record under the name:

_acme-challenge.mail.mysuperfakedomain.gov.

with the following value:

cmV1AljDcvVJjcv-0hgOAAvWt3CJaHtuS8G3b_ss3zw

- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
Press Enter to Continue

- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
Please deploy a DNS TXT record under the name:

_acme-challenge.mysuperfakedomain.gov.

with the following value:

SuDA0J4HDFK_pNEX7CfI5yLJi0LgYL9NkdTY3UDDu8s

(This must be set up in addition to the previous challenges; do not remove,
replace, or undo the previous challenge tasks yet. Note that you might be
asked to create multiple distinct TXT records with the same name. This is
permitted by DNS standards.)

- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
Press Enter to Continue

- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
Please deploy a DNS TXT record under the name:

_acme-challenge.www.mysuperfakedomain.gov.

with the following value:

UMVtRnz1mC4Sc_eWHCe6gebyNiuYyug27kDp-0qnO4k

(This must be set up in addition to the previous challenges; do not remove,
replace, or undo the previous challenge tasks yet. Note that you might be
asked to create multiple distinct TXT records with the same name. This is
permitted by DNS standards.)

Before continuing, verify the TXT record has been deployed. Depending on the DNS 
provider, this may take some time, from a few seconds to multiple minutes. You can
check if it has finished deploying with aid of online tools, such as the Google
Admin Toolbox: https://toolbox.googleapps.com/apps/dig/#TXT/_acme-challenge.www.mysuperfakedomain.gov.
Look for one or more bolded line(s) below the line ';ANSWER'. It should show the
value(s) you've just added.

- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
Press Enter to Continue
Waiting for verification...
Cleaning up challenges

IMPORTANT NOTES:
 - Congratulations! Your certificate and chain have been saved at:
   /etc/letsencrypt/live/mysuperfakedomain.gov/fullchain.pem
   Your key file has been saved at:
   /etc/letsencrypt/live/mysuperfakedomain.gov/privkey.pem
   Your certificate will expire on 2021-08-11. To obtain a new or
   tweaked version of this certificate in the future, simply run
   certbot again. To non-interactively renew *all* of your
   certificates, run "certbot renew"
 - If you like Certbot, please consider supporting our work by:

   Donating to ISRG / Let's Encrypt:   https://letsencrypt.org/donate
   Donating to EFF:                    https://eff.org/donate-le

Syncing certbot configs to my-certificates-bucket
upload: ../../etc/letsencrypt/archive/mysuperfakedomain.gov/cert1.pem to s3://my-certificates-bucket/archive/mysuperfakedomain.gov/cert1.pem
upload: ../../etc/letsencrypt/archive/mysuperfakedomain.gov/chain1.pem to s3://my-certificates-bucket/archive/mysuperfakedomain.gov/chain1.pem
upload: ../../etc/letsencrypt/archive/mysuperfakedomain.gov/privkey1.pem to s3://my-certificates-bucket/archive/mysuperfakedomain.gov/privkey1.pem
upload: ../../etc/letsencrypt/archive/mysuperfakedomain.gov/fullchain1.pem to s3://my-certificates-bucket/archive/mysuperfakedomain.gov/fullchain1.pem
upload: ../../etc/letsencrypt/csr/0203_csr-certbot.pem to s3://my-certificates-bucket/csr/0203_csr-certbot.pem
upload: ../../etc/letsencrypt/keys/0203_key-certbot.pem to s3://my-certificates-bucket/keys/0203_key-certbot.pem
upload: ../../etc/letsencrypt/live/mysuperfakedomain.gov/fullchain.pem to s3://my-certificates-bucket/live/mysuperfakedomain.gov/fullchain.pem
upload: ../../etc/letsencrypt/live/mysuperfakedomain.gov/chain.pem to s3://my-certificates-bucket/live/mysuperfakedomain.gov/chain.pem
upload: ../../etc/letsencrypt/live/mysuperfakedomain.gov/privkey.pem to s3://my-certificates-bucket/live/mysuperfakedomain.gov/privkey.pem
upload: ../../etc/letsencrypt/live/mysuperfakedomain.gov/cert.pem to s3://my-certificates-bucket/live/mysuperfakedomain.gov/cert.pem
upload: ../../etc/letsencrypt/renewal/mysuperfakedomain.gov.conf to s3://my-certificates-bucket/renewal/mysuperfakedomain.gov.conf
upload: ../../etc/letsencrypt/live/mysuperfakedomain.gov/README to s3://my-certificates-bucket/live/mysuperfakedomain.gov/README
```

**Review by dav3r — APPROVED** [review]
Manual override- what a useful feature, thanks for adding it!  ⏏️

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -60,6 +60,9 @@ repos:
     rev: 1.0.5
     hooks:
       - id: shell-lint
+        # TODO: SC2039 is retired in 0.7.2 but not 0.7.0.  Local checks will
+        # pass but they will currently fail in GitHub Actions which uses 0.7.0
+        args: [-x, --exclude=SC2039]
```
Comment (jsf9k) [inline .pre-commit-config.yaml]: Can you create an issue to remove this when possible and include a link to it here?  One of the checkboxes in the standard PR template references doing this, so it is required.

**Review by jsf9k — CHANGES_REQUESTED** [review]
I requested one minor change, and then I will heartily approve.  Please re-request my approval after addressing it.

File: .pre-commit-config.yaml (hunk 1 above)
Comment (felddy) [reply .pre-commit-config.yaml]: At @mcdonnnj 's suggestion I've implemented this differently so that it won't require a disable or a TODO.

Comment (felddy) [reply .pre-commit-config.yaml]: Some additional context about `SC2039`: 
https://github.com/koalaman/shellcheck/wiki/SC2039

And an issue about `SC30xx` codes not being properly reported:
https://github.com/koalaman/shellcheck/issues/2052

**Comment by felddy** [comment]
@dav3r : Just saw `mysuperfakedomain`  teehee

File: .pre-commit-config.yaml (hunk 1 above)
Comment (jsf9k) [reply .pre-commit-config.yaml]: @felddy - normally we let the reporter resolve comments.

File: src/entrypoint.sh [hunk 2]
```diff
@@ -4,7 +4,7 @@ set -o nounset
 set -o errexit
 # Sha-bang cannot be /bin/bash (not available), but
 # the container's /bin/sh does support pipefail.
-# shellcheck disable=SC2039
+# shellcheck disable=SC2039,SC3040
```
Comment (jsf9k) [inline src/entrypoint.sh:12]: What do you think about adding [these links](https://github.com/cisagov/certboto-docker/pull/22#discussion_r632088000) to this comment?

**Review by jsf9k — APPROVED** [review]
Approved, but @felddy please see my one suggestion.

**Review by mcdonnnj — APPROVED** [review]
A solid addition. 💪💪💪

File: src/entrypoint.sh (hunk 2 above)
Comment (felddy) [reply src/entrypoint.sh:12]: Good suggestion.  Added in https://github.com/cisagov/certboto-docker/pull/22/commits/37276f7a78f38a9cff62dba5c9a410b08943161e

**Pull request merged**